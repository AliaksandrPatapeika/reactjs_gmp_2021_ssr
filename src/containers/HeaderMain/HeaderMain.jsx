import './HeaderMain.less';

import React from 'react';

import Search from '../Search';

const HeaderMain = () => (
  <div className="headerMainContainer">
    <h1 className="title">FIND YOUR MOVIE</h1>
    <Search />
  </div>
);

export default HeaderMain;
