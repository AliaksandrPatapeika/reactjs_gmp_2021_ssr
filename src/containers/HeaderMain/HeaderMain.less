.headerMainContainer {
	display: flex;
	flex-direction: column;
	padding: 30px 66px 158px;
}
