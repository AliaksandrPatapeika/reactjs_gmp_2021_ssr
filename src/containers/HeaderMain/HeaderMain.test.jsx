import React from 'react';
import {Provider} from 'react-redux';
import renderer from 'react-test-renderer';
import configureMockStore from 'redux-mock-store';

import HeaderMain from './HeaderMain';

jest.mock('next/router', () => ({
  useRouter: () => ({
    query: {search: 'movie'}
  })
}));

describe('HeaderMain', () => {
  test('renders correctly', () => {
    const mockStore = configureMockStore();
    const initialState = {
      query: {
        filter: '',
        sortBy: 'release_date',
        sortOrder: 'desc'
      }
    };
    const store = mockStore(initialState);
    const headerMain = renderer
      .create(
        <Provider store={store}>
          <HeaderMain />
        </Provider>
      )
      .toJSON();

    expect(headerMain).toMatchSnapshot();
  });
});
