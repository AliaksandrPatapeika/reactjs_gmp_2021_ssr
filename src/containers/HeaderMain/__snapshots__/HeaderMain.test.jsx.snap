// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderMain renders correctly 1`] = `
<div
  className="headerMainContainer"
>
  <h1
    className="title"
  >
    FIND YOUR MOVIE
  </h1>
  <div
    className="searchContainer"
  >
    <input
      autoComplete="off"
      className="searchInput"
      onChange={[Function]}
      placeholder="What do you want to watch?"
      value=""
    />
    <button
      className="searchButton"
      disabled={false}
      id={null}
      onClick={[Function]}
      type="button"
    >
      SEARCH
      
    </button>
  </div>
</div>
`;
